### DISCUSSION TITLE
429 error for refresh token

### DISCUSSION THREAD

**Discussion by bowu** [supabase/supabase#16640] [Questions]
Hi,

My nextjs app uses Supabase and it works well on mac (both chrome and safari) and linux (both firefox and chrome). However, it keeps having problems with Windows (both Edge and Chrome). The app oftentimes sends requests to PROJECT_URL/auth/v1/token?grant_type=refresh_token and such requests may succeed or fail (429 error). When they fail, the app will experience a bunch of 401 errors. Does anyone have an idea about what's going on?

On mac or linux, it never sends redundant refresh_token requests.

**Comment by GaryAustin1** [comment]
Have you looked at the network tab at the requests to supabase server?
Also you could look in the dashboard logs, API Edge log and Auth log likely being the most useful.
Is there in error information when you get the 429 error.  It implies you generated to many auth requests too quickly.

**Reply by bowu** [reply]
Yes, I have. You are right that the app generated too many auth requests (i.e. refresh token requests) too frequently. I'm confused because it only happens on Windows. The network requests log seems normal on mac and linux.

**Reply by GaryAustin1** [reply]
You probably have to post some code and more detail for someone who knows nextjs to to help.  I personally would do that in Supabase's Discord as you might get more interactive help than here.  You could also make sure you have latest supabase-js and auth-helpers, if you are using that.  I don't see any recent reports of this multiple token refresh issue in Discord or here.  Some debug to narrow down what is causing the multiple requests is needed.  The refresh timer is usually set based on the jwt expiration time.

**Comment by leo4life2** [comment]
having this too

**Comment by songjeongjun320** [comment]
Is there a someone who resolve this? Just waiting?

**Reply by GaryAustin1** [reply]
I would post your details in a new post or use Discord.
Could be lots of things and very dependent on your code.
I don't know of any bugs involved with this type of thing, so most likely some coding issue in your client code.

**Comment by entermask** [comment]
same
